File: CodingInterviewsPackSolutionsCPP/Valid Anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        unordered_map<char,int>freq1,freq2;
        for(int i = 0; i < s.size(); i++)
            freq1[s[i]]++;
        for(int i = 0; i < t.size(); i++)
            freq2[t[i]]++;
        if(freq1.size()!=freq2.size())
            return false;
        
        for(const pair<char,int>&p:freq1){
            char ch = p.first;
            int freq = p.second;
            
            if(freq!=freq2[ch])
                return false;
        }
        return true;
    }
};

File: CodingInterviewsPythonSolutions/Valid Anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        freq1 = {}
        freq2 = {}
        
        for ch in s:
            freq1[ch] = freq1.get(ch, 0) + 1
        
        for ch in t:
            freq2[ch] = freq2.get(ch, 0) + 1
        
        if len(freq1) != len(freq2):
            return False
        
        for ch, freq in freq1.items():
            if freq != freq2.get(ch, 0):
                return False
        
        return True
